codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsProtocolUtils.java

```diff
@@ -126,7 +126,8 @@ static void generateXmlParseBody(GenerationContext context) {
                     writer.write("const parsedObj = xmlParse(encoded, { attributeNamePrefix: '', "
                             + "ignoreAttributes: false, parseNodeValue: false, tagValueProcessor: (val, tagName) "
                             + "=> decodeEscapedXML(val) });");
-                    writer.write("return parsedObj[Object.keys(parsedObj)[0]];");
+                    writer.write("const key = Object.keys(parsedObj)[0];");
+                    writer.write("return { [key]: parsedObj[key][\"#text\"], ...parsedObj[key] };");
```

Comment:
This will result in an undefined value for the wrapper if `"#text"` isn't set and in a the field staying around if it is set.

```js
// Without "#text" set:
{
   "foo": undefined,
   "bar": { ... }
}

// With "#text" set:
{
   "foo": "contents",
   "#text": "contents"
}
```

Not sure if this will cause issues anywhere else, but we should clean it up just in case.

Reply:
Can you provide a concrete example like the one explained in https://github.com/aws/aws-sdk-js-v3/pull/993#issuecomment-597220541?

Reply:
[Examples of behavior here.](https://typescript-play.js.org/#code/MYewdgzgLgBAtiAJgSwGYE8YF4YAoAOAhgE4QCmiA8gEYBWAXDIWOgJSPOZYB8MA3gCgYwmKEiwA1mS4watMsCgA6KeggES5KnVYBtAAwBdANxCRxMlACuxMPxi7VhxkVIU5j6Yd0AiAMRQZAAeUD6GADQwStGuWh5OMAC+pokCAgD06TAAciAwAcFQAmIQIAA2ZEplIADmuAgoGLiCIjA+AGIgID6MLa0iPgBCJD1t1CNmwqmJrKymmTC8fJ0gjFZgiGSoyGAUkcPEjD7jxD6pGVkAEoQQ+YEhxeClFVW19Uho6M2TbSujff02gUQqNjhNWtNZqYLot+CsjicfJFgVAESNEkA)

Reply:
The example:
```json
{
  "Foo": {
    "Bar": "bar"
  }
}
```
is same as <AccessControlPolicy> in https://github.com/aws/aws-sdk-js-v3/pull/993#issuecomment-597220541
`Foo` is same as `AccessControlPolicy`, and is not what we're looking for. The values in `Owner` and `DisplayName` are important which are available.

The example:
```json
{
  "Foo": {
    "#text": "bar"
  }
}
```

is same as <LocationConstraint> one in https://github.com/aws/aws-sdk-js-v3/pull/993#issuecomment-597220541
`#text` will be present in the output, but will not be consumed anywhere

Reply:
Fixed in commit https://github.com/aws/aws-sdk-js-v3/pull/993/commits/4c3d0ee28e46997023ad669a01d1d5cc8cc5e436
Tested in [playground](https://typescript-play.js.org/#code/MYewdgzgLgBAtiAJgSwGYE8YF4YAoAOAhgE4QCmiA8gEYBWAXDIWOgJSPOZYB8MA3gCgYwmKEiwoZAB5QAckjKzCcMthgAiAMSSZ6gNxCRY6DADWZLjBq0ywKADpz6CARLkqdVgG0ADAF0DEVFwEyJSCmsAFRAAJTIoAFdiMDUw92svJwDDYTQ8NIi6aLjE5K8dOQUlFT9WfhygmAKPWmL4pLBMiz9Ut0LW2PayivlERWUybMaRMYAbeNVmqMHSzpGqiamRAF8G4iGUpaKVjoNdgQB6C5h5GG1pKAFjEHn7WZAAc1wEFAxcQSC6gAYiAQOpGADphoAEIkcEaahwhq7basVgGK4wXh8WHERjqRHEdTnTEACUIEDuFSeIReZDen2+SDQ6H+DWBoPhkOmWgq8IJSKCKLRBku12xIJA+MJxKAA)